File: source_funcs/go_transcoder/MINIMUM_XOR_VALUE_PAIR/MINIMUM_XOR_VALUE_PAIR_emit.go
package main

import "math"

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

//go:export f_gold
func f_gold(arr []int, n int) int {
	var min_xor int = math.MaxInt64
	for i := int(0); i < n; i++ {
		for j := int(i + 1); j < n; j++ {
			min_xor = min(min_xor, arr[i]^arr[j])
		}
	}
	return min_xor
}

File: rust_funcs/cpp_transcoder/MINIMUM_XOR_VALUE_PAIR/MINIMUM_XOR_VALUE_PAIR_emit.rs
#![allow(dead_code)]
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(unused_mut)]
#![allow(non_snake_case)]
#![allow(non_camel_case_types)]
#![allow(improper_ctypes_definitions)]
#[no_mangle]
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
#[no_mangle]
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
#[no_mangle]
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
#[no_mangle]
fn len(arr: &[i32]) -> usize {arr.len()}
#[no_mangle]
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32; 

#[no_mangle]
fn f_gold(arr: &mut [i32], n: i32) -> i32 {
    let mut min_xor = f32::MAX as i32;
    for i in 0..n {
        for j in (i + 1)..n {
            min_xor = min_xor.min(arr[i as usize] ^ arr[j as usize]);
        }
    }
    min_xor
}
